datamonster_api/lib/datamonster.py

```diff
@@ -160,6 +163,40 @@ def get_datasource_by_id(self, datasource_id):
         datasource["uri"] = self._get_datasource_path(datasource_id)
         return self._datasource_result_to_object(datasource, has_details=True)
 
+    def get_datasource_details(self, datasource_id):
+        """Get details (metadata) for the given datasource
+
+        :param datasource_id: The ID of the datasource for which we get the details
+        :return: dictionary object with the datasource details
+        """
+        return self.client.get(self._get_datasource_path(datasource_id))
+
+    def _get_datasource_path(self, datasource_id):
+        return "{}/{}".format(self.datasource_path, datasource_id)
+
+    def _get_rawdata_path(self, datasource_id, params):
+        return self.rawdata_path.format(
+            self._get_datasource_path(datasource_id),
+            six.moves.urllib.parse.urlencode(params),
+        )
+
+    def _get_dimensions_path(self, uuid):
+        return self.dimensions_path.format(uuid)
+
+    def _datasource_result_to_object(self, datasource, has_details=False):
+        ds_inst = Datasource(
```

Comment:
This looks like it might not play well with your other PR, which passes arguments to these constructors as param dictionaries. Not necessarily a problem, just noting that you may have to address merge conflicts

Reply:
yes, this is an older PR - i will handle merge conflicts on https://github.com/adaptivemgmt/datamonster-api/pull/11

Reply:
word